exp/api/v1alpha3/machinepool_types.go

```diff
@@ -25,7 +25,7 @@ import (
 
 const (
 	// MachinePoolFinalizer is used to ensure deletion of dependencies (nodes, infra).
-	MachinePoolFinalizer = "machinepool.exp.cluster.x-k8s.io"
```

Comment:
If we don't want to change the old-api, we can disable conversion webhooks here.

Reply:
Should we mention that this finalizer needs to be converter as well in case folks just convert the APIGroup? 

Otherwise, we can end up with some MachinePools still using this finalizer that will never get deleted

Reply:
We could, although that might be a bit confusing, and the controller or conversion webhook has to take care of the conversion, maybe we export both?

Reply:
Did you mean something like this @vincepri @yastij https://github.com/kubernetes-sigs/cluster-api/pull/4574/commits/a86f1da0e6bca3ba0630d3d999f30a59ad9203e9 ?